manifests/module.pp

```diff
@@ -138,13 +138,17 @@
   }
 
   # CONFIGURE LNET
-  $configure_lustre_command = "modprobe lnet && lnetctl lnet configure && \
-    lnetctl import ${lnet_conf_file} && \
-    echo 'If this file is deleted Puppet will reconfigure LNet' > ${lnet_trigger_file}"
+
+  ## when LNet is (re)configured, write a trigger file containing the last reboot time
+  $configure_lustre_command = "umount -a -t lustre && lustre_rmmod && \
+    modprobe lnet && lnetctl lnet configure && lnetctl import ${lnet_conf_file} && \
+    echo 'If this file is deleted or contains an older system boot time Puppet will unmount Lustre and reconfigure LNet' > ${lnet_trigger_file} && \
+    uptime -s >> ${lnet_trigger_file}"
+
+  ## configure LNet, but only if the trigger file is missing or has a different last reboot time
   exec { 'configure_lustre':
     command => $configure_lustre_command,
-    creates => $lnet_trigger_file,
-    unless  => "test -f ${lnet_trigger_file}",
+    unless  => "test -f ${lnet_trigger_file} && egrep -q \"$(uptime -s)\" ${lnet_trigger_file}",
```

Comment:
I don't think you still need to test if the `lnet_trigger_file` exists. I'm pretty sure the egrep command will fail if the file doesn't exist. It doesn't hurt to leave both configured, but it does make the unless command a bit harder to read.

Reply:
You're right. I've updated to remove the unnecessary check for file and retested the trigger conditions on a Magnus node.